Code/GraphMol/Depictor/Templates.cpp

```diff
@@ -0,0 +1,90 @@
+//
+//  Copyright (C) 2023 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "Templates.h"
+
+namespace RDDepict {
+bool CoordinateTemplates::isValidTemplate(RDKit::ROMol& mol, const std::string& smiles) {
+    // template must have 2D coordinates
+    if (mol.getNumConformers() == 0) {
+        BOOST_LOG(rdWarningLog) << "WARNING: Template is missing coordinates: "
+                                << smiles << "\n";
+        return false;
+    }
+    if (mol.getConformer().is3D()) {
+        BOOST_LOG(rdWarningLog) << "WARNING: Template has 3D coordinates, 2D "
+                                << "coordinates required: " << smiles << "\n";
+        return false;
+    }
+
+    // Make sure this template is a single ring system (spiro'd ring systems are OK).
```

Comment:
This doesn't technically check that there's only one ring system: `c1ccccc1-c1ccccc1` would pass the test below.
I think checking that every bond is a ring bond would do what you want.

Reply:
You're totally right, meant to do bonds! I will update that